afpacket/afpacket.go

```diff
@@ -248,16 +250,17 @@ func (h *TPacket) releaseCurrentPacket() error {
 func (h *TPacket) ZeroCopyReadPacketData() (data []byte, ci gopacket.CaptureInfo, err error) {
 	h.mu.Lock()
 retry:
-	if h.current == nil || !h.current.next() {
+	if h.current == nil || !h.headerNextNeeded || !h.current.next() {
 		if h.shouldReleasePacket {
 			h.releaseCurrentPacket()
 		}
 		h.current = h.getTPacketHeader()
```

Comment:
Hmm, maybe past-me isn't as dumb as he looks.  Let me take another close look at this change, and I'll try to get back to you in a day or two.